src/pal/src/cruntime/misc.cpp

```diff
@@ -634,3 +637,33 @@ void MiscUnsetenv(const char *name)
     }
     InternalLeaveCriticalSection(pthrCurrent, &gcsEnvironment);
 }
+
+#if defined(_DEBUG)
+
+/*++
+Function:
+PAL_memcpy
+
+Overlapping buffer-safe version of memcpy.
+See MSDN doc for memcpy
+--*/
+void *PAL_memcpy (void *dest, const void *src, size_t count)
+{
+    UINT_PTR x = (UINT_PTR)dest, y = (UINT_PTR)src;
+    void *ret;
+    //if the buffers do not overlap, use memcpy
+    if((x + count <= y) || (y + count <= x))
+    {
+        #define memcpy_dummy memcpy
+        #undef memcpy
+        ret = memcpy(dest, src, count);
+        #define memcpy memcpy_dummy
+    }
+    //Otherwise, use memmove since it's overlap-safe 
+    else 
+    {
+        ret = memmove(dest, src, count);
```

Comment:
We should not fallback to memmove here. It should be an assert instead. The goal is to find places in the codebase that are using memcpy incorrectly and fix them.